src/core/api/api_error.pl

```diff
@@ -166,6 +166,35 @@
                               'api:reason' : Err },
              'api:message' : Msg
             }.
+api_global_error_jsonld(error(document_not_found(Id), _), Type, JSON) :-
+    error_type(Type, Type_Displayed),
+    format(string(Msg), "Document not found: ~q", [Id]),
+    JSON = _{'@type' : Type_Displayed,
+             'api:status' : "api:not_found",
+             'api:error' : _{ '@type' : 'api:DocumentNotFound',
+                              'api:document_id' : Id },
+             'api:message' : Msg
+            }.
+api_global_error_jsonld(error(document_not_found(Id, Document), _), Type, JSON) :-
```

Comment:
Do we actually generate this in a way that it appears at this level of error transform? document api endpoints should be transforming this to include the document.

Reply:
This is what `call_catch_document_mutation` does, it checks for `known_document_error` and if the case is included, it makes sure to insert the document it was currently inserting in there. Is that appropriate in this case?

Reply:
Yeah, this confused me for a bit. So, `document_not_found` does not actually appear in `known_document_error`.